### PR TITLE
Scalar loads/stores shouldn't invalidate predicate vectorization

### PR DESCRIPTION
**Pull request by vksnk**

### CODE REVIEW

**Review by dsharletg — APPROVED** [review]
Nice find. This explains why it was working in some of my specializations and not others: the specializations were for broadcasting (scalar loads).

**Comment by steven-johnson** [comment]
Does this need a test?

**Comment by steven-johnson** [comment]
Looks like an error in correctness_predicated_store_load: https://buildbot.halide-lang.org/master/#/builders/126/builds/334

**Comment by vksnk** [comment]
Fixed the failure and added a test to verify that the change does something useful (also verified that the new test failed before the change).

File: test/correctness/predicated_store_load.cpp [hunk 1]
```diff
@@ -91,6 +91,31 @@ int predicated_tail_test(const Target &t) {
     return 0;
 }
 
+int predicated_tail_with_scalar_test(const Target &t) {
+    int size = 73;
+    Var x("x"), y("y");
+    Func f("f"), g("g";
```
Comment (steven-johnson) [inline test/correctness/predicated_store_load.cpp:97]: Missing a `)` here

Comment (vksnk) [reply test/correctness/predicated_store_load.cpp]: Oops, last minute clean-up has gone wrong.

**Comment by steven-johnson** [comment]
Another correctness_predicate_store_load failure: https://buildbot.halide-lang.org/master/#/builders/126/builds/346

**Comment by vksnk** [comment]
> Another correctness_predicate_store_load failure: https://buildbot.halide-lang.org/master/#/builders/126/builds/346

Sorry, having some issues with set up of Hexagon SDK, so couldn't verify locally. 

It seems that at some point for some of the tests predicated loads/stores stopped to appear and this PR fixes that, so some of the test values needed to be adjusted.

**Comment by vksnk** [comment]
Bots are green, submitting.

**Comment by steven-johnson** [comment]
Tagging @alexreinking, this might be needed for a future backport to 12.0.2 (not sure how urgent)

**Comment by alexreinking** [comment]
> Tagging @alexreinking, this might be needed for a future backport to 12.0.2 (not sure how urgent)

We _can_ do this, but maybe it's worth waiting for a few more bugfixes, given how much buildbot time a release takes?

**Comment by dsharletg** [comment]
I'm not sure how necessary backporting this is. It was a long-standing performance bug...

**Comment by alexreinking** [comment]
OTOH - long-standing performance bugs that don't break ABI (like this one) are good candidates for backporting.

**Pull request merged**